js/indicators/aroon.src.js

```diff
@@ -0,0 +1,181 @@
+    /**
+     * Aroon. This series requires the `linkedTo` option to be
+     * set and should be loaded after the `stock/indicators/indicators.js`.
+     *
+     * @extends plotOptions.sma
+     * @product highstock
+     * @sample {highstock} stock/indicators/aroon
+     *                     Aroon
+     * @since 7.0.0
+     * @excluding
+     *             allAreas,colorAxis,compare,compareBase,joinBy,keys,stacking,
+     *             showInNavigator,navigatorOptions,pointInterval,
+     *             pointIntervalUnit,pointPlacement,pointRange,pointStart
+     * @optionparent plotOptions.aroon
+     */
+    {
+        /**
+         * Paramters used in calculation of aroon series points.
+         * @excluding periods, index
+         */
+        params: {
+            /**
+             * Period for Aroon indicator
+             *
+             * @default 14
+             * @type {number}
+             * @since 7.0.0
+             * @product highstock
+             */
+            period: 14
+        },
+        marker: {
+            enabled: false
+        },
+        tooltip: {
+            pointFormat: '<span style="color:{point.color}">\u25CF</span><b> {series.name}</b><br/>Aroon Up: {point.y}<br/>Aroon Down: {point.aroonDown}<br/>'
+        },
+        /**
+         * aroonDown line options.
+         *
+         * @since 7.0.0
+         * @product highstock
+         */
+        aroonDown: {
+            /**
+             * Styles for an aroonDown line.
+             *
+             * @since 7.0.0
+             * @product highstock
+             */
+            styles: {
+                /**
+                 * Pixel width of the line.
+                 *
+                 * @type {Number}
+                 * @since 7.0.0
+                 * @product highstock
+                 */
+                lineWidth: 1,
+                /**
+                 * Color of the line. If not set, it's inherited from
+                 * [plotOptions.aroon.color](
+                 * #plotOptions.aroon.color).
+                 *
+                 * @type {Highcharts.ColorString}
+                 * @since 7.0.0
+                 * @product highstock
+                 */
+                lineColor: undefined
+            }
+        },
+        dataGrouping: {
+            approximation: 'averages'
+        }
+    }, /** @lends Highcharts.Series.prototype */ H.merge(multipleLinesMixin, {
+        nameComponents: ['period'],
+        nameBase: 'Aroon',
+        pointArrayMap: ['y', 'aroonDown'],
+        pointValKey: 'y',
+        linesApiNames: ['aroonDown'],
+        getValues: function (series, params) {
+            var period = params.period,
+                xVal = series.xData,
+                yVal = series.yData,
+                yValLen = yVal ? yVal.length : 0,
+                AR = [], // 0- date, 1- Aroon Up, 2- Aroon Down
+                xData = [],
+                yData = [],
+                slicedY,
+                low = 2,
```

Comment:
I assume low and high indexes are fixed in algorithm. In that case we should check if `yData` is an array of numbers (for reference see RSI).

Reply:
Aroon indicator is a difference between low or high value and period. So it can be calculated also for a simple line chart. Low or high indexes are not required.

Reply:
Thanks!